### ISSUE TITLE
[Test Infrastructure] Replace Mock Sampling Strategy with Real Implementation in Generation Fixtures

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/BitNet-rs#378]
# Problem Description

The test infrastructure contains a mock sampling strategy that uses a simple argmax approach instead of a proper probabilistic sampling implementation. This limits the quality and realism of generation tests and may miss edge cases that occur with real sampling strategies.

## Environment

- **File**: `tests-new/fixtures/fixtures/generation_fixtures.rs`
- **Function**: `mock_sample_from_logits` (lines 462-472)
- **Crate**: Test fixtures for generation testing
- **Component**: Test infrastructure / Generation fixtures

## Root Cause Analysis

The `mock_sample_from_logits` function in the generation fixtures uses a deterministic argmax approach:

```rust
async fn mock_sample_from_logits(&self, logits: &[f32]) -> Result<u32> {
    // Find argmax for deterministic behavior
    let max_idx = logits
        .iter()
        .enumerate()
        .max_by(|(_, a), (_, b)| a.partial_cmp(b).unwrap())
        .map(|(idx, _)| idx)
        .unwrap_or(0);

    Ok(max_idx as u32)
}
```

The comment explicitly states "replace with real implementation", indicating this is temporary scaffolding code that should be replaced with proper sampling logic.

## Impact Assessment

**Severity**: Medium
**Priority**: Normal

**Affected Components**:
- Generation test fixtures
- Temperature scaling tests
- Sampling strategy validation
- Test data generation quality

**Impact**:
- Test fixtures generate unrealistic data using only greedy sampling
- Temperature scaling tests don't reflect real sampling behavior
- Missing coverage for probabilistic sampling edge cases
- Reduced confidence in generation pipeline validation

## Current Behavior vs Expected Behavior

**Current**: Mock sampling always selects the token with highest logit value (greedy sampling)
**Expected**: Real probabilistic sampling that respects temperature, top-k, top-p, and other sampling parameters

## Proposed Solution

Replace the mock implementation with the real `SamplingStrategy` from `crates/bitnet-inference/src/sampling.rs`:

### Primary Implementation

```rust
use crate::sampling::{SamplingStrategy, SamplingConfig};

async fn sample_from_logits(&self, logits: &[f32], config: Option<SamplingConfig>) -> Result<u32> {
    let sampling_config = config.unwrap_or_else(|| SamplingConfig {
        temperature: 1.0,
        top_k: 0,
        top_p: 1.0,
        repetition_penalty: 1.0,
        seed: Some(42), // Deterministic for tests
    });
    
    let mut strategy = SamplingStrategy::new(sampling_config);
    strategy.sample(logits, &[])
}
```

### Configuration Support

Add sampling configuration to generation fixtures:

```rust
#[derive(Debug, Clone)]
pub struct GenerationFixtureConfig {
    pub sampling_config: SamplingConfig,
    pub temperature_range: Vec<f32>,
    pub top_k_values: Vec<u32>,
    pub top_p_values: Vec<f32>,
    // ... other config fields
}
```

### Enhanced Test Cases

```rust
async fn generate_sampling_test_data(&mut self) -> Result<()> {
    let logits = vec![2.0, 1.0, 0.5, 0.1];
    
    // Temperature scaling tests with real sampling
    for &temperature in &[0.1, 0.5, 1.0, 1.5, 2.0] {
        let config = SamplingConfig {
            temperature,
            seed: Some(42),
            ..Default::default()
        };
        
        let sampled_token = self.sample_from_logits(&logits, Some(config)).await?;
        self.sampling_test_data.temperature_tests.push((temperature, vec![sampled_token]));
    }
    
    // Top-k sampling tests
    for &top_k in &[1, 5, 10, 50] {
        let config = SamplingConfig {
            top_k,
            seed: Some(42),
            ..Default::default()
        };
        
        let sampled_token = self.sample_from_logits(&logits, Some(config)).await?;
        self.sampling_test_data.top_k_tests.push((top_k, vec![sampled_token]));
    }
    
    // Top-p sampling tests
    for &top_p in &[0.1, 0.5, 0.9, 1.0] {
        let config = SamplingConfig {
            top_p,
            seed: Some(42),
            ..Default::default()
        };
        
        let sampled_token = self.sample_from_logits(&logits, Some(config)).await?;
        self.sampling_test_data.top_p_tests.push((top_p, vec![sampled_token]));
    }

    Ok(())
}
```

### Phase 1: Integration Setup
- [ ] Add dependency on `bitnet-inference` crate to test fixtures
- [ ] Import `SamplingStrategy` and `SamplingConfig` types
- [ ] Update `Cargo.toml` for test workspace dependencies

### Phase 2: Core Implementation
- [ ] Replace `mock_sample_from_logits` with real sampling implementation
- [ ] Add sampling configuration support to fixture generation
- [ ] Ensure deterministic behavior for reproducible tests

### Phase 3: Enhanced Test Coverage
- [ ] Add comprehensive sampling strategy test cases
- [ ] Implement temperature, top-k, and top-p test scenarios  
- [ ] Add repetition penalty test cases
- [ ] Validate sampling distribution properties

### Phase 4: Validation & Integration
- [ ] Update existing tests that depend on the mock implementation
- [ ] Ensure backward compatibility of generated test data
- [ ] Add regression tests for sampling behavior
- [ ] Update documentation for test fixture usage

### Unit Tests
```rust
#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_deterministic_sampling() {
        let fixtures = GenerationFixtures::new().await.unwrap();
        let logits = vec![1.0, 2.0, 0.5];
        
        let config = SamplingConfig {
            seed: Some(42),
            temperature: 1.0,
            ..Default::default()
        };
        
        let token1 = fixtures.sample_from_logits(&logits, Some(config.clone())).await.unwrap();
        let token2 = fixtures.sample_from_logits(&logits, Some(config)).await.unwrap();
        
        assert_eq!(token1, token2, "Sampling should be deterministic with same seed");
    }
    
    #[tokio::test]
    async fn test_temperature_effect() {
        let fixtures = GenerationFixtures::new().await.unwrap();
        let logits = vec![10.0, 1.0, 1.0]; // Heavy bias toward first token
        
        // Low temperature should prefer high-probability tokens
        let low_temp_config = SamplingConfig {
            temperature: 0.1,
            seed: Some(42),
            ..Default::default()
        };
        
        // High temperature should be more random
        let high_temp_config = SamplingConfig {
            temperature: 2.0,
            seed: Some(43), // Different seed for variety
            ..Default::default()
        };
        
        let low_temp_token = fixtures.sample_from_logits(&logits, Some(low_temp_config)).await.unwrap();
        let high_temp_token = fixtures.sample_from_logits(&logits, Some(high_temp_config)).await.unwrap();
        
        // Low temperature should strongly prefer token 0
        assert_eq!(low_temp_token, 0);
    }
}
```

### Integration Tests
- Validate generated test data quality
- Ensure sampling strategies work with different vocabulary sizes
- Test edge cases (empty logits, invalid probabilities)
- Performance benchmarks for test data generation

## Risk Assessment

**Low Risk Changes**:
- Replacing mock with real implementation maintains same interface
- Deterministic seeding preserves test reproducibility

**Medium Risk Changes**:
- Generated test data will change (requires test fixture regeneration)
- Potential performance impact on test execution time

**Mitigation Strategies**:
- Implement with feature flag for gradual rollout
- Maintain backward compatibility mode during transition
- Comprehensive regression testing before deployment

## Dependencies

**Internal**:
- `crates/bitnet-inference/src/sampling.rs` - Core sampling implementation
- Test workspace configuration updates

**External**:
- `rand` crate (already in use)
- `rand_chacha` crate (already in use)

## Acceptance Criteria

- [ ] `mock_sample_from_logits` function replaced with real sampling implementation
- [ ] Deterministic test behavior preserved through seeding
- [ ] Support for all sampling strategies (temperature, top-k, top-p, repetition penalty)
- [ ] Comprehensive test coverage for sampling behavior
- [ ] Performance impact within acceptable bounds (< 10% test execution time increase)
- [ ] All existing tests pass with new implementation
- [ ] Documentation updated to reflect real sampling usage

## Related Issues

This issue addresses technical debt in the test infrastructure and complements:
- Production sampling strategy implementation (already complete in `sampling.rs`)
- Generation pipeline testing improvements
- Test fixture quality enhancements

## References

- **Implementation**: `crates/bitnet-inference/src/sampling.rs` - Complete sampling strategy implementation
- **Test file**: `tests-new/fixtures/fixtures/generation_fixtures.rs:462-472`
- **Documentation**: BitNet.rs sampling strategies documentation
